apis/go.mod

```
@@ -1,14 +1,16 @@
 module github.com/metal3-io/baremetal-operator/apis
 
-go 1.22
+go 1.22.0
+
+toolchain go1.22.4
 
 require (
 	github.com/google/uuid v1.6.0
 	github.com/metal3-io/baremetal-operator/pkg/hardwareutils v0.5.1
 	github.com/stretchr/testify v1.9.0
-	k8s.io/api v0.29.9
-	k8s.io/apimachinery v0.29.9
-	sigs.k8s.io/controller-runtime v0.17.6
+	k8s.io/api v0.30.1
```

```suggestion
	k8s.io/api v0.30.5
```
Bump all the k8s directy to `v0.30.5`  to avoid dependabot bump right away.

---

thats a followup I will do if dependabot doesn't pick it up, I was thinking to do it but then I opted not to since it has nothing to do with the fix of the issue.